.github/workflows/ci.yml

```diff
@@ -87,6 +87,41 @@ jobs:
         if: steps.cache.outputs.cache-hit != 'true' && matrix.cache == 'true'
         run: rm -rf ~/.m2/repository/io/trino/trino-*
 
+  reproducible-build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # checkout all commits, as the build result depends on `git describe` equivalent
+          ref: |
+            ${{ github.event_name == 'repository_dispatch' &&
+                github.event.client_payload.pull_request.head.sha == github.event.client_payload.slash_command.args.named.sha &&
+                format('refs/pull/{0}/head', github.event.client_payload.pull_request.number) || '' }}
+      - uses: ./.github/actions/setup
+        with:
+          cache: false # cache not used, dependencies will be downloaded to `maven.repo.local`
+          cleanup-node: true # not enough space to fetch dependencies and build twice
+      - name: First build to a local repository
+        # ~/.m2/repository isn't used because we are using `maven.repo.local` to store all downloaded dependencies and generated artifacts
```

Comment:
How about moving this above, so that it's shared
```yaml
# ~/.m2/repository isn't used because we are using `maven.repo.local` to store all downloaded dependencies and generated artifacts
- name: First build to a local repository
  run: ...
- name: Second build to a local repository
  run: ...